Create Stack.py

shirapahmer opened the pull request:
This is a Stack class that runs with drawable.py instead of the recordclass library
We noticed a problem (with visualizationApp.py we think) that if an invalid number is entered as an argument in the animation, the Operations grid moves over a few spaces to the right and only moves back when a valid argument is entered.

JMCanning78 commented on PythonVisualizations/Stack.py:
```diff
@@ -0,0 +1,208 @@
+        toPositions = (self.cellCoords(len(self.list)), 
+                       self.cellCenter(len(self.list)))
+
+        # Animate arrival of new value from operations panel area
```
The comment here was probably copied from Array.py.  The animation for Stack has the new values sliding in from the top of the canvas.

JMCanning78 commented:
Regarding the possible [bug you mentioned](https://github.com/JMCanning78/datastructures-visualization/pull/36#issue-436506064) "if an invalid number is entered as an argument in the animation, the Operations grid moves over a few spaces to the right and only moves back when a valid argument is entered", it's the result of the Tk geometry manager and the way the control panel is set up in the VisualizationApp.
The Operations panel is placed to be centered above the animation speed control and the status message.  When the invalid number is put in the text entry area, the status message is filled with a longer error message "Input value must be an integer from 0 to 99."  That longer message expands the Tk Label widget beyond its minimum width and that makes the control panel wider.  Tk re-centers the operations panel which is what offsets its position.
There are many ways of changing that behavior.  I'm not sure that any need to be explored just yet since more operations are going to be added to Stack, which will change the size of the operations area.  Making changes to the control panel in VisualizationApp will impact all the visualizations, so we'll need to decide what the desired behavior is for all of them.

shirapahmer commented:
> The work you've done on the Stack.py program went well beyond what was in Issue #17. You've done most of the work in Issue #29 by incorporating the VisualizationApp framework. There are a couple of things that should be changed to accomplish both Issues (including the additional feature in #29 of adding a pointer for the "top" of the stack. I think the best way to do that will be to ask you to address both tasks on Issue #29 and submit a new PR.
> 
> The new Stack.py replaces the realStack.py program in the repo. The new PR should either
> 
> 1. rename realStack.py as Stack.py, or
> 2. delete realStack.py while adding Stack.py.
> 
> Option 1 can be done either with the `git` command line tool or on GitHub.com (https://help.github.com/en/github/managing-files-in-a-repository/renaming-a-file), but not with GitHub Desktop. Option 2 can be done by all 3, although it somewhat hides the fact that the file had a different name in earlier versions.

We made a new branch to address Issue 29 and created a pull request. We are hesitant to change the name

shirapahmer commented:
I accidentally closed this and don't know how to open it again, sorry. But I think we did address the issues. We opened a new branch for Issue 29 and submitted a pull request for it, and then we renamed RealStack to Stack and changed the files. We weren't so confident that we were doing the right thing but we think we may have figured it out.

shirapahmer commented:
We opened a new branch for Issue 29 and submitted a pull request for it, and then we renamed RealStack to Stack and changed the files. We weren't so confident that we were doing the right thing but we think we may have figured it out.

JMCanning78 commented:
I'm closing this PR because PR #58 replaces it.
